packages/task/src/browser/task-service.ts

```diff
@@ -409,6 +409,40 @@ export class TaskService implements TaskConfigurationClient {
     }
 
     async runTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
+        const runningTasksInfo: TaskInfo[] = await this.getRunningTasks();
+
+        // check if the task is active
+        const matchedRunningTaskInfo = runningTasksInfo.find(taskInfo => {
+            const taskConfig = taskInfo.config;
+            return this.taskDefinitionRegistry.compareTasks(taskConfig, task);
+        });
+        if (matchedRunningTaskInfo) { // the task is active
+            const taskName = this.taskNameResolver.resolve(task);
+            const taskId = matchedRunningTaskInfo.taskId;
+            const terminalId = matchedRunningTaskInfo.terminalId;
+            let terminal: TerminalWidget | undefined;
+            if (terminalId) {
+                terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
+                if (terminal) {
+                    terminal.show(); // bring the terminal to the top
+                }
+            }
+            this.messageService.info(`The task \'${taskName}\' is already active`, 'Terminate Task', 'Restart Task').then(actionTask => {
+                if (actionTask) {
+                    this.taskServer.kill(taskId);
+                    if (actionTask === 'Restart Task' && terminal) {
+                        terminal.close();
+                        this.doRunTask(task, option);
+                    }
+                }
+            });
+            return;
+        } else { // run task as the task is not active
+            return this.doRunTask(task, option);
+        }
+    }
+
+    private async doRunTask(task: TaskConfiguration, option?: RunTaskOption): Promise<TaskInfo | undefined> {
```

Comment:
Can we have `protected` modifier here?
I think it will be more flexible for cases when some class extends `TaskService`
From this point of view also would nice to have at least `protected` `restartTask` method.
For example, at the moment `attach` method is `public` and it's possible to override this logic, provide own logic. 
`restartTask` logic is relative to it.
wdyt?